### ISSUE TITLE
Issue with restriction in some cases

### ISSUE THREAD

**Issue by SamSalvaggio** [bozenne/BuyseTest#11]
Dear Brice, 

I'm sorry to bother you again. I have found a bug in the package, still present in version "3.0.0".

It seems to be linked to using two non time to event variables followed by a restricted TTE variable in the hierarchy. Note that if you add any kind of variable before 2 non tte, it crashes as well. It seems to be linked with ending the formula with this pattern of 3 subsequent variables : any combinaison of 2 non tte and 1 restricted tte.

Here is a reproducible example of the bug : 

```
set.seed(1)
dt <- simBuyseTest(n.T = 50, n.C = 50,
                   names.strata = "strat_column", n.strata = 2,
                   argsTTE = list(name = c("tte1"), 
                                  name.censoring = c("cnsr1"),
                                  scale.T = c(200), scale.censoring.T = c(10^5),
                                  scale.C = c(200), scale.censoring.C = c(10^5)),
                   argsBin = list(p.T = list(c(0.4,0.6))),
                   argsCont = list(mu.T = c(0.3), sigma.T = 1)
                  )

formula1 <- treatment ~ strat_column +
  bin(toxicity, operator = "<0") +
  cont(score, threshold = 0.5) +
  tte(tte1, status = cnsr1, threshold = 10, restriction = 365)

GPC <- BuyseTest(formula1,
                    data = dt,
                    trace = FALSE)

```
Thanks in advance for your help.
Kind regards,
Samuel

**Comment by bozenne** [comment]
Dear Samuel,

Thanks for the bug report. I could see that some indexing went wrong internally when processing the data for the restricted case. It should be solved now (version 3.0.1).

Best regards
Brice

**Comment by SamSalvaggio** [comment]
Thanks a lot Brice :)
